SwiftUIHW/SwiftUIHW/ContentView.swift

```diff
@@ -7,21 +7,194 @@
+                volumeSlider
+                shareDownloadButtons
+            }
+        }
+    }
+    
+    // MARK: - Private properties
+    
+    @ObservedObject private var viewModel = PlayerViewModel()
+    
+    @State private var volumeValue: Float = Constants.zero
+    @State private var songDuration: Float = Constants.zero
+    @State private var isPlaying = false
+    @State private var isDownloaded = false
+    @State private var isShareAlertShowed = false
+    
+    private var volumeSlider: some View {
+        Slider(value: Binding(get: {
+            viewModel.currentVolume
+        }, set: { newValue in
+            viewModel.currentVolume = newValue
+            viewModel.updateVolume(value: newValue)
+        }), in: Constants.zero...Constants.hundrid) {
+            
+        } minimumValueLabel: {
+            Image(systemName: Constants.lowSpeakerImageName).foregroundColor(.white)
+        } maximumValueLabel: {
+            Image(systemName: Constants.highSpeakerImageName).foregroundColor(.white)
+        }.padding([.leading, .trailing], Constants.sliderPaddingValue)
+    }
+    
+    private var songTime: some View {
+        HStack {
+            Spacer()
+            Text("\(viewModel.songTime())").padding([.bottom, .trailing], Constants.sliderPaddingValue)
+        }
+    }
+    
+    private var gradientBackround: some View {
+        LinearGradient(gradient: Gradient(colors: [Color(Constants.colorName), Color(Constants.gradientColorName)]), startPoint: .top, endPoint: .bottom).ignoresSafeArea()
+    }
+    
+    private var songProgressSlider: some View {
+        Slider(value: Binding(get: {
+            viewModel.currentTime
+        }, set: { newValue in
+            viewModel.currentTime = newValue
+            viewModel.setTime(value: newValue)
+        }), in: Constants.zero...viewModel.maxDuration).padding([.leading, .trailing], Constants.sliderPaddingValue)
+    }
+    
+    private var shareDownloadButtons: some View {
+        HStack {
+            Button {
+                isShareAlertShowed = true
+            } label: {
+                Image(systemName: Constants.shareImageName).resizable()
+                    .frame(width: Constants.buttonImageWidth, height: Constants.buttonImageHeight)
+                    .foregroundColor(.white)
+            }.alert(isPresented: $isShareAlertShowed) {
+                Alert(title: Text(Constants.shareQuestion), primaryButton: .default(Text(Constants.yesText)), secondaryButton: .cancel())
+            }
+            Spacer()
+            Button {
+                isDownloaded = true
+            } label: {
+                Image(systemName: Constants.downloadImageName).resizable()
+                    .frame(width: Constants.buttonImageWidth, height: Constants.buttonImageHeight)
+                    .foregroundColor(.white)
+            }.confirmationDialog(Text(""), isPresented: $isDownloaded) {} message: {
```

Comment:
```suggestion
            }.confirmationDialog(Text(""), isPresented: $isDownloaded) {} message: {
```
пустые строки вроде тоже в константы выносим